resolver/resolver_test.go

```diff
@@ -309,127 +84,168 @@ func TestNonMesosHandler(t *testing.T) {
-	if err != nil {
-		t.Error(err)
-	}
-	var got2 records.Config
-	err = json.Unmarshal(g2, &got2)
-	eq2 := reflect.DeepEqual(got2, res.config)
-	if !eq2 {
-		t.Error("HTTP config API failure")
-	}
-
-	// test /v1/services -- existing record
-	r3, err := http.Get(ts.URL + "/v1/services/_leader._tcp.mesos.")
-	if err != nil {
-		t.Error(err)
-	}
-	g3, err := ioutil.ReadAll(r3.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got3 []map[string]interface{}
-	err = json.Unmarshal(g3, &got3)
-	correct3 := []map[string]interface{}{{"host": "leader.mesos.", "port": "5050", "service": "_leader._tcp.mesos.", "ip": "1.2.3.4"}}
-	eq3 := reflect.DeepEqual(got3, correct3)
-	if !eq3 {
-		t.Error("HTTP services API failure")
-	}
-
-	// test /v1/services -- non existing record
-	r4, err := http.Get(ts.URL + "/v1/services/_myservice._tcp.mesos.")
-	if err != nil {
-		t.Error(err)
-	}
-	g4, err := ioutil.ReadAll(r4.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got4 []map[string]interface{}
-	err = json.Unmarshal(g4, &got4)
-	correct4 := []map[string]interface{}{{"host": "", "port": "", "service": "", "ip": ""}}
-	eq4 := reflect.DeepEqual(got4, correct4)
-	if !eq4 {
-		t.Error("HTTP services API failure")
-	}
-
-	// test /v1/host -- existing record
-	r5, err := http.Get(ts.URL + "/v1/hosts/leader.mesos")
-	if err != nil {
-		t.Error(err)
-	}
-	g5, err := ioutil.ReadAll(r5.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got5 []map[string]interface{}
-	err = json.Unmarshal(g5, &got5)
-	correct5 := []map[string]interface{}{{"host": "leader.mesos.", "ip": "1.2.3.4"}}
-	eq5 := reflect.DeepEqual(got5, correct5)
-	if !eq5 {
-		t.Error("HTTP hosts API failure")
+	for _, tt := range []struct {
+		path      string
+		code      int
+		got, want interface{}
+	}{
+		{"/v1/version", http.StatusOK, map[string]interface{}{},
+			map[string]interface{}{
+				"Service": "Mesos-DNS",
+				"URL":     "https://github.com/mesosphere/mesos-dns",
+				"Version": "0.1.1",
+			},
+		},
+		{"/v1/config", http.StatusOK, &records.Config{}, &res.config},
+		{"/v1/services/_leader._tcp.mesos.", http.StatusOK, []interface{}{},
+			[]interface{}{map[string]interface{}{
+				"service": "_leader._tcp.mesos.",
+				"host":    "leader.mesos.",
+				"ip":      "1.2.3.4",
+				"port":    "5050",
+			}},
+		},
+		{"/v1/services/_myservice._tcp.mesos.", http.StatusOK, []interface{}{},
+			[]interface{}{map[string]interface{}{
+				"service": "",
+				"host":    "",
+				"ip":      "",
+				"port":    "",
+			}},
+		},
+		{"/v1/hosts/leader.mesos", http.StatusOK, []interface{}{},
+			[]interface{}{map[string]interface{}{
+				"host": "leader.mesos.",
+				"ip":   "1.2.3.4",
+			}},
+		},
+	} {
+		if resp, err := http.Get(ts.URL + tt.path); err != nil {
+			t.Error(err)
+		} else if got, want := resp.StatusCode, tt.code; got != want {
+			t.Errorf("GET %s: StatusCode: got %d, want %d", tt.path, got, want)
+		} else if err := json.NewDecoder(resp.Body).Decode(&tt.got); err != nil {
```

Comment:
Not really readable. Needs some seconds to find the `Decode(&tt.got)` with side-effect on `tt.got`.